spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -260,36 +261,53 @@ public AcknowledgeMode getAcknowledgeMode() {
 	 */
 	public void setQueueNames(String... queueName) {
 		Assert.noNullElements(queueName, "Queue name(s) cannot be null");
-		this.queueNames = new CopyOnWriteArrayList<>(queueName);
+		setQueues(Arrays.stream(queueName)
+				.map(n -> new Queue(n))
+				.collect(Collectors.toList())
+			.toArray(new Queue[0]));
 	}
 
 	/**
 	 * Set the name of the queue(s) to receive messages from.
 	 * @param queues the desired queue(s) (can not be <code>null</code>)
 	 */
 	public final void setQueues(Queue... queues) {
-		setQueueNames(collectQueueNames(queues));
-	}
-
-	private static String[] collectQueueNames(Queue... queues) {
 		Assert.notNull(queues, "'queues' cannot be null");
 		Assert.noNullElements(queues, "'queues' cannot contain null elements");
-		String[] queueNames = new String[queues.length];
-		for (int i = 0; i < queues.length; i++) {
-			queueNames[i] = queues[i].getName();
+		if (isRunning()) {
+			for (Queue queue : queues) {
```

Comment:
? queues is `Queue[]`. It would have to be `Arrays.stream(queues).forEach()`; for loop is simpler.

https://stackoverflow.com/questions/35518471/in-java-8-why-were-arrays-not-given-the-foreach-method-of-iterable